packages/frontend/src/features/sqlRunner/store/lineChartSlice.ts

```diff
@@ -0,0 +1,64 @@
+import { ChartKind, deepEqual, isLineChartSQLConfig } from '@lightdash/common';
+import { createSlice } from '@reduxjs/toolkit';
+import { SqlRunnerResultsTransformerFE } from '../transformers/SqlRunnerResultsTransformerFE';
+import { cartesianChartConfigSlice } from './cartesianChartBaseSlice';
+import { setSavedChartData, setSqlRunnerResults } from './sqlRunnerSlice';
+
+export const lineChartConfigSlice = createSlice({
+    name: 'lineChartConfig',
+    initialState: cartesianChartConfigSlice.getInitialState(),
+    reducers: {
+        ...cartesianChartConfigSlice.caseReducers,
+    },
+    extraReducers: (builder) => {
+        builder.addCase(setSqlRunnerResults, (state, action) => {
+            if (action.payload.data.results && action.payload.data.columns) {
+                const sqlRunnerResultsTransformer =
+                    new SqlRunnerResultsTransformerFE({
+                        rows: action.payload.data.results,
+                        columns: action.payload.data.columns,
+                    });
+                if (action.payload.data.columns) {
+                    state.options = {
+                        xLayoutOptions:
+                            sqlRunnerResultsTransformer.cartesianChartXLayoutOptions(),
+                        yLayoutOptions:
+                            sqlRunnerResultsTransformer.cartesianChartYLayoutOptions(),
+                        groupByOptions:
+                            sqlRunnerResultsTransformer.cartesianChartGroupByLayoutOptions(),
+                    };
+                }
+
+                // Update layout
+                const oldDefaultLayout = state.defaultLayout;
+                const newDefaultLayout =
+                    sqlRunnerResultsTransformer.defaultCartesianChartLayout();
+                state.defaultLayout = newDefaultLayout;
+
+                if (
+                    !state.config ||
+                    deepEqual(
+                        oldDefaultLayout || {},
+                        state.config?.fieldConfig || {},
+                    )
+                ) {
+                    state.config = {
+                        metadata: {
+                            version: 1,
+                        },
+                        type: ChartKind.LINE,
+                        fieldConfig: newDefaultLayout,
+                        display: state.config?.display,
+                    };
+                }
+            }
+        });
+        builder.addCase(setSavedChartData, (state, action) => {
+            if (isLineChartSQLConfig(action.payload.config)) {
+                state.config = action.payload.config;
+            }
+        });
+    },
+});
+
+export type lineChartActionsType = typeof lineChartConfigSlice.actions;
```

Comment:
I believe we want to export this from `lineChartSlice`

Reply:
And could we have it Pascal case?

Reply:
> I believe we want to export this from lineChartSlice

What? 😅 

Happy to make this Pascal case

Reply:
> I believe we want to export this from lineChartSlice 

local changes not being up to date ^ ignore!